src/components/product.js

```diff
@@ -91,16 +94,16 @@ export default class Product extends Component {
    * @return {Boolean}
    */
   get shouldUpdateImage() {
-    return !this.cachedImage || (this.image && this.image.src && this.image.src !== this.cachedImage.src);
+    return !this.cachedImage || (this.image && this.image.src !== this.cachedImage);
   }
 
   /**
    * get image for product and cache it. Return caches image if shouldUpdateImage is false.
    * @return {Object} image objcet.
    */
   get currentImage() {
-    if (this.shouldUpdateImage) {
-      this.cachedImage = this.model.images[0];
+    if (this.shouldUpdateImage && this.image) {
```

Comment:
isn't the check for `this.image` redundant here? And if not, should it not be put in the `shouldUpdateImage` predicate?